core/src/main/java/org/infinispan/util/CoreBlockHoundIntegration.java

```diff
@@ -133,10 +134,13 @@ private static void questionableMethodsAllowedToBlock(BlockHound.Builder builder
          throw new CacheException(e);
       }
 
-      // This method calls readScopedState which reads scope from a file that
-      // can block the current thread while doing I/O
+      // Can wait on lock
+      builder.allowBlockingCallsInside(ClusterTopologyManagerImpl.class.getName(), "updateState");
+
+      // These methods to I/O via GlobalStateManager
```

Comment:
@wburns I've been thinking about excluding all the `GlobalStateManagerImpl` methods here instead of their callers, WDYT?

Reply:
Sure, more specific is better :)

Reply:
I will take a look a the methods to see if we should have a JIRA or not to convert the methods instead of just an exclusion.

Reply:
Sure if just excluding the file reading at `GlobalStateManagerImpl` works, then I am all for it.